modules/amazon/index.js

```diff
@@ -1,17 +1,58 @@
 const XRay = require('x-ray');
 const xray = XRay();
 
+const amazonUrlWhitelist = [
+  "amazon.com",
+  "amazon.de",
+  "amazon.co.uk",
+  "amazon.co.jp",
+  "amazon.cn",
+  "amzn.eu",
+  "amzn.asia",
+  "amzn.to",
+  "a.co/",
+];
+
+const isAmazonUrl = function(url) {
+  for (var amazUrl in amazonUrlWhitelist) {
+    if (url.includes(amazUrl)) return true;
+  }
+  return false;
+};
+
 module.exports.url = function(url, reply) {
   var query = {
-    price: '#priceblock_ourprice',
-    product_name: '#productTitle'
+    ourPrice: '#priceblock_ourprice',
+    dealPrice: '#priceblock_dealprice',
+    salePrice: '#priceblock_saleprice',
+    offerPrice: '.offer-price',
+    productName: '#productTitle'
   };
-  if (url.includes("amazon")) {
+  if (isAmazonUrl(url)) {
     xray(url, query)(function(err, data) {
       if (err) reply(err);
-      var productName = data.product_name.replace('\n', ' ').trim();
-      var price = data.price;
-      reply(`${productName} - [${price}]`);
+      if (typeof data.productName === "undefined") {
+        reply("Error: Failed to find product name");
+        return;
+      }
+      var productName = data.productName.replace('\n', ' ').trim();
+      var ourPrice = data.ourPrice;
```

Comment:
I think this could more succinctly be written as:

```
let pric e= Object.keys(query).map((selector) data[selector]).filter((price) => price !== undefined)[0]
```